src/ServicePulse.Host/app/js/views/monitored-endpoints/monitored-endpoints.controller.js

```diff
@@ -13,12 +13,9 @@
         });
 
         function updateUI() {
-            monitoringService.getRaw().then(function (data) {
-                $scope.endpoints = data["NServiceBus.Endpoints"];
-
-                timeoutId = $timeout(function () {
-                    updateUI();
-                }, 5000);
+            monitoringService.getEndpoints().subscribe(function (endpoint) {
+                $scope.endpoints = $scope.endpoints || {};
+                $scope.endpoints[endpoint.endpointName] = endpoint;
```

Comment:
I much prefer passing each endpoint individually. The consuming UI can then evaluate each response on the merits of that specific endpoint without having to know which service it came from. This way the controller doesn't need to know anything about the number of monitoring services involved, which should keep the code simpler. We can also in the future use the same observable to listen for updates to specific endpoints (maybe use the RxJs `filter` method?) and only send those updates through to the controller instead of sending all endpoints to the controller again.

The alternative is to have code in the UI controller to iterate the list of endpoints returned and then only evaluate those endpoints. What benefit are you seeing of doing it that way?

To change it to pass the arrays though it's as simple as removing the `selectMany` from the `getEndpoints` method, so it's completely trivial to achieve.